feature-utils/poly-look/src/react-components/overlays/sideSlider.jsx

```diff
@@ -0,0 +1,422 @@
+ * defaults to false.
+ * @param {callback} [props.onClose] - Called after the exit animations end.
+ * @param {string} [props.leftDistance] - Contents distance from the left
+ * of the screen. Can be any value that is compatible with CSS calc rule;
+ * defaults to 124px.
+ * @param {Array[number]} [props.backdropColor] - The backdrop color as a rgba array;
+ * defaults to [0, 0, 0, 0.3].
+ * @param {string} [props.screenMaxWidth] - The max width of the screen;
+ * defaults to the value of --max-width.
+ * @param {string} [props.animationDuration] - The duration of the animations.
+ * Can be any value that is compatible with CSS transition-duration rule;
+ * defaults to 0.6s.
+ * @param {JSX.Element} [props.Component] - The component that will be animated.
+ * If this component can trigger the SideSlider to close then it must have
+ * an onClose callback as one of it's props.
+ * If it has clickable elements, then Event.stopPropagation()
+ * must be called in the event's handler function, otherwise the Slider will close.
+ * If the component has vertical scroll then it must accept custom styles.
+ * @param {string} [props.lastChildSelector] - CSS selector for the last child of
+ * the Component prop. Use it if you think the Component might have
+ * a vertical scroll, unnecessary otherwise.
+ * @returns {JSX.Element}
+ */
+
+const SideSlider = ({
+  open = false,
+  onClose,
+  leftDistance = "124px",
+  backdropColor = [0, 0, 0, 0.3],
+  screenMaxWidth = "var(--max-width)",
+  animationDuration = "0.6s",
+  Component,
+  lastChildSelector,
+}) => {
+  let [red, green, blue, alpha] = backdropColor;
+  let commonStyles = {
+    duration: { transitionDuration: `${animationDuration}` },
+    noInteraction: { pointerEvents: "none" },
+    interaction: { pointerEvents: "auto" },
+    backdropOut: { backgroundColor: `rgba(${red}, ${green}, ${blue}, 0)` },
+    backdropIn: {
+      backgroundColor: `rgba(${red}, ${green}, ${blue}, ${alpha})`,
+    },
+    contentsIn: { transform: `translateX(${leftDistance})` },
+    contentsOutRight: { transform: `translateX(${screenMaxWidth})` },
+    backgroundTransition: { transition: "background-color" },
+    contentsTransition: { transition: "transform" },
+    noScroll: { overflowY: "hidden" },
+  };
+
+  const [step, updateStep] = useState(null);
+  const [isPressing, updateMousePress] = useState(false);
+  const [contentsStyle, updateContentsStyle] = useState(
+    commonStyles.contentsOutRight
+  );
+  const [backdropStyle, updateBackdropStyle] = useState({});
+  const [verticalGesture, updateVerticalGesture] = useState("scroll");
+  const [childScroll, updateChildScroll] = useState({});
+  const [movement, updateMovement] = useState({
+    startX: null,
+    startY: null,
+    currentX: null,
+    currentY: null,
+    swipeY: null,
+    direction: null,
+  });
+  const [axis, setAxis] = useState("");
+  const [animationsDoneCount, updateAnimationsStatus] = useState(0);
```

Comment:
Maybe use the same wording in the two variables, as anywhere else?

Reply:
Done.